src/Agent.Worker/Build/TrackingManager.cs

```diff
@@ -147,27 +144,7 @@ private void WriteToFile(string file, object value)
 
             // Create the directory if it does not exist.
             Directory.CreateDirectory(Path.GetDirectoryName(file));
-
-            // Serialize the object.
-            JsonSerializerSettings settings = new JsonSerializerSettings();
-            settings.ContractResolver = new CamelCasePropertyNamesContractResolver();
-            settings.Converters.Add(new StringEnumConverter() { CamelCaseText = true });
-            string contents = JsonConvert.SerializeObject(
-                value: value,
-                formatting: Formatting.Indented,
-                settings: settings);
-
-            // Lock write access on the file so the garbage collector service
-            // can also lock on write access to safely handle the file.
-            using (FileStream stream = new FileStream(file, FileMode.Create, FileAccess.Write, FileShare.Read))
-            {
-                using (StreamWriter writer = new StreamWriter(stream))
-                {
-                    writer.Write(contents);
-                    writer.Flush();
-                    stream.Flush();
-                }
-            }
+            IOUtil.SaveObject(value, file);
```

Comment:
Can you revert this or fix it to reconcile the differences between the existing code and the IOUtil method? You're effectively changing the encoding of how this file is written. I don't want to start writing these files in a totally different encoding for no reason, with old files already written on disk in a different encoding. The old Windows agent should upgrade cleanly to this new agent, and it's files will be written in the old encoding. I'm also uncertain whether the file share mode (see comment in the source code) is same when switching over to the IOUtil helper. Did you confirm that?